### PR TITLE
Resolved issue #15 and added true iterator for efficient reading of records

### PR DESCRIPTION
**Pull request by thammegowda**
+ The issue #15 is resolved. `slice()` method is now proper
+ `RecordIterator`, a real iterator in java is added to efficiently read records( the subsequent calls to `slice()` has heavy disk seek overhead as it has to skip records till the start position for each call).
+ Test case  `SequenceReaderTest` is added

### CODE REVIEW

**Comment by chrismattmann** [comment]
great work thanks @thammegowda

File: examples/seq_reader.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 data = nutchpy.sequence_reader.slice(5,20,path)
 # print(data)
-assert len(data) == 2
+assert len(data) == 3
```
Comment (ahmadia) [inline examples/seq_reader.py]: Why is a slice from 5 to 20 yielding a length of 3?

File: nutchpy/readers.py [hunk 2]
```diff
@@ -209,21 +210,19 @@ def _rec_to_doc(self, rec):
...
-        max = self.count(path)
-        while start < max:
-            batch = self.slice(start, start + batch_size, path)
-            for item in batch:
-                yield self._rec_to_doc(item) if auto_convert else item
-            start = start + batch_size
+        rec_iterator = self.reader.getRecordIterator(path, start)
+        while rec_iterator.hasNext():
+            rec = rec_iterator.next()
+            yield self._rec_to_doc(list(rec)) if auto_convert else rec
```
Comment (ahmadia) [inline nutchpy/readers.py]: looks good.

File: nutchpy/readers.py [hunk 3]
```diff
@@ -209,21 +210,19 @@ def _rec_to_doc(self, rec):
...
         '''
-        Reads the sequence file records by slicing batch by batch.
-        This is a streaming reader, unlike reader.read(...) which loads whole file into memory at once
+        Creates a stream of (key:value) pairs by reading given sequence file
+        This is a streaming reader, unlike reader.read(...) which loads whole file contents
+         into memory at once
         :param path: path of the file
         :param start: the start position, default is 0
-        :param batch_size: back size default is 20
-        :return: iterator of records
```
Comment (ahmadia) [inline nutchpy/readers.py]: is the auto_convert flag undocumented?

File: seqreader-app/src/main/java/com/continuumio/seqreaderapp/RecordIterator.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
+package com.continuumio.seqreaderapp;
```
Comment (ahmadia) [inline seqreader-app/src/main/java/com/continuumio/seqreaderapp/RecordIterator.java]: nicely done.

**Comment by ahmadia** [comment]
This looks really good.  Thanks @thammegowda

File: examples/seq_reader.py (hunk 1 above)
Comment (thammegowda) [inline examples/seq_reader.py]: Since the example data has 8 records `[0..7]`, the `slice(5, 20)` function returns last three records `[5,6,7]`.

**Comment by thammegowda** [comment]
:+1:

**Pull request merged**